migration-guide.md

```diff
@@ -0,0 +1,57 @@
+
+### Swift 2 → Swift 3
+
+**IMPORTANT:** Make granular commits after each step and substep in this guide. Make sure the commit messages follow the proper rules.
+
+1. Clone project and pull/checkout all branches ([script here](http://stackoverflow.com/a/21189710/1001803)).
+    - Clean up all branches and keep only `master` and `develop`
+    - Make sure `develop` is even or ahead of `master`
+2. From `develop` branch run `git checkout -b feature/swift3-migration`
+3. *(optional)* Clean project, remove unused files, update file structure.
+4. Update `Cartfile` to point to Swift 3 versions of dependencies  
+    - Update framework names if used in project
+        - `NStack` → `NStackSDK`
+        - `Serializable` → `Serpent`
+    - Remove existing Carthage folder and run `carthage update --platform ios` (change platform as needed)
+5. *(optional)* Update translations generator 
+    - Download from latest Swift 3 release [on Github](https://github.com/nodes-ios/nstack-translations-generator/releases), do NOT add to Xcode
+    - Remove any old translations generator and templates present in project
+6. Open `YourProject.xcodeproj` in Xcode 8 and run Swift migrator
+    - Go through suggested changes to double check
+7. Go to Build Phases and modify current run scripts
+    - Update translations script as per [README](https://github.com/nodes-ios/nstack-translations-generator)
+    - Add or update script checking for TODO & FIXME
+    - Add or update Carthage copy frameworks script with new names/frameworks
+8. Update build settings
+    - Enable whole module optimizations for release/adhoc
+    - Enable bitcode if possible
+    - Remove unused/old custom keys
+9. Build and fix errors, rinse & repeat until project compiles 
+    - **Alamofire** 
+	    - Update completion type to `DataResponse`
+	    - Rename completion closure paramter to response instead of result
+	    - Switch on `response.result` instead of result
+	    - Update switch cases to `.success(let data)` and `.failure(let error)`
+    - **Serpent**
+	    - Replace `import Serializable` with `import Serpent`
+	    - Remove `completionHandler:` named parameter in Alamofire requests
+	    - (**important**) Set default unwrapper, or manually specify per request
+    - **Blobfish**
+	    - Completely replace ErrorHandler or add error handling if not present
+	    - Properly setup blobs for token expired, connection error, unknown error
+    - and more *(read documentation of frameworks)*
+    - *(optional)* Refactor code to newer standards along the way (small changes)
+10. Fix as many (ideally all) warnings in the project, so that it's clean and updated 
+11. *(optional)* If using HockeySDK, add `NSPhotoLibraryUsageDescription` to the `Info.plist`
+12. Add or udpate README
+    - Change Swift / Xcode versions
+    - Small project description
+    - Things to watch out for
+    - Add test users if necessary 
+
```

Comment:
The guide started with steps related to git, it would be nice to end in a similar way. Something like:
13. When everything is done, merge `feature/swift3-migration` into `development`
14. Delete `feature/swift3-migration`
15. (?) should `develop` be merged into `master`? If it's a framework, probably yes and a new major release should be made. If it's an app, probably not. `develop` should be merged into `master` when a new version is uploaded on the App Store